src/pages/Swap/Swap.tsx

```diff
@@ -95,9 +136,48 @@ export const Swap: FC = () => {
       approxAmountA: balanceB.price / balanceA.price
     }
   }, [balance, selectedTokenA, selectedTokenB])
-  const handleSwap = () => {
-    console.log('ere bois')
+
+  const handleSwap = async () => {
+    try {
+      const txBuilder = createTransactionBuilder()
+      const tx = await swapTokens(
+        amountTokenA,
+        selectedTokenA,
+        selectedTokenB,
+        userPublicKey,
+        userSourceTokenType,
+        userTargetTokenType,
+        slippage,
+        GammaProgram,
+        connection
+      )
+      txBuilder.add(tx)
+      setSendingTransaction(true)
+      // eslint-disable-next-line max-len
+      const sourceAmount = `${bigNumberFormatter(new BigNumber(amountTokenA))} ${selectedTokenA?.symbol}`
+      // eslint-disable-next-line max-len
+      const targetAmount = `${bigNumberFormatter(new BigNumber(amountTokenB))} ${selectedTokenB?.symbol}`
+      const { success, txSig } = await sendTransaction(txBuilder, {
+        successMessage: `You successfully swapped ${sourceAmount} to ${targetAmount}`
+      })
+      console.log('SwapResponse', success)
+      if (!success) {
+        //off(connectionId)
+        console.log('An error occurred while Swapping!')
+        setSendingTransaction(false)
+        return
+      } else {
+        setSendingTransaction(false)
+        setAmountTokenA('')
+        setAmountTokenB('')
+        await forceCronAndUpdateLocalData(txSig)
+      }
+    } catch (e) {
+      setSendingTransaction(false)
+      console.log('An error occurred while depositing.', e)
+    }
   }
+
   // TODO: these values pls bois
   const impactPercent = 0.1
   const gfxFee = 0.0005 + 0.0005
```

Comment:
Same with this, its supposed to be network + gfx fee; we should intheory know the network fee but do we know what we are going to set the gfx fee to?